Silicon airlock hotkey screen tips

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

- Adds support for screen tips for cyborgs and AIs
- Adds support for screen tips for shift-click, alt-click and ctrl-shift-click interactions
- Adds screen tips for cyborgs and AI airlock interactions
- ⚠️Moves screen tips down a tile ⚠️ so they don't get cut off when you have more than two lines' worth
- Makes the "LMB: Open" door screen tip for living mobs only show when they're adjacent to them (I don't think I'm going to be the one to make it check for edge cases like TK)

https://user-images.githubusercontent.com/25089914/154612360-44242e99-b8c2-4993-a51f-50734cdee857.png
https://user-images.githubusercontent.com/25089914/154612366-aedeb88c-e919-441d-97bf-ae4b3fbbe470.png
https://user-images.githubusercontent.com/25089914/154614182-dff65c1e-5736-4546-83fb-c9c715ca932e.png
https://user-images.githubusercontent.com/25089914/154614187-7282ef45-4d14-4091-ab8e-c0cc3b2b1489.png

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Demystifies airlock hotkeys for newer silicon players (I know you can just examine but they don't know that)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
qol: Screen tips for cyborg and AI airlock hotkeys
qol: The "LMB: Open" screentip for doors will only show when adjacent (no, it doesn't factor in things like TK)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
AI door screen tips for maximising door interaction efficiency

cacogen committed:
Moves screentips down a tile in a bold move

Mothblocks commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -627,9 +627,20 @@
+			context[SCREENTIP_CONTEXT_SHIFT_LMB] = density ? "Open" : "Close"
+			context[SCREENTIP_CONTEXT_CTRL_LMB] = locked ? "Unbolt" : "Bolt"
+			context[SCREENTIP_CONTEXT_ALT_LMB] = isElectrified() ? "Unelectrify" : "Electrify"
+			context[SCREENTIP_CONTEXT_CTRL_SHIFT_LMB] = emergency ? "Unset EA" : "Set EA"
```
I'm guessing you've already tried "emergency access" and it was too long, but the abbreviation here does make me sad, since new players won't know what it means. ☹️

Mothblocks replied:
Not that I have any better answers here, if anyone does reply here

cacogen replied:
![Space Station 13 2022-02-18 170425](https://user-images.githubusercontent.com/25089914/154615564-6fe4c77a-dcca-4e7c-a6d2-947453b85ff2.png)

![Space Station 13 2022-02-18 170430](https://user-images.githubusercontent.com/25089914/154615530-c967cedb-aa18-4cd9-893f-8e06ad3240ec.png)

Think it's probably fine actually
️

cacogen committed:
Challenge everything

Mothblocks commented on code/game/atoms.dm:
```diff
@@ -2129,19 +2132,33 @@
+								shift_lmb_ctrl_shift_lmb_line += " | "
+							shift_lmb_ctrl_shift_lmb_line += "[SCREENTIP_CONTEXT_CTRL_SHIFT_LMB]: [context[SCREENTIP_CONTEXT_CTRL_SHIFT_LMB]]"
+
+						extra_context = "<br><span style='font-size: 7px'>[lmb_rmb_line][ctrl_lmb_alt_lmb_line][shift_lmb_ctrl_shift_lmb_line]</span>"
```
Can you make this only happen if any of these exist?

cacogen committed:
Makes extra_context only set if it has context

- Allows AI camera mobs to see contextual screentips
- Fixes comment referring to wrong shortcut

cacogen committed:
Do not move screentip position

cacogen committed:
Holy calamity scream insanity all you ever gonna be's another great fan of me, break

Mothblocks commented on code/game/atoms.dm:
```diff
@@ -2140,14 +2143,32 @@
+						if(extra_lines)
+							extra_context = "<br><span style='font-size: 7px'>[lmb_rmb_line][ctrl_lmb_alt_lmb_line][shift_lmb_ctrl_shift_lmb_line]</span>"
+							active_hud.screentip_text.maptext_y = -10 + (extra_lines - 1) * -9	//first extra line pushes atom name line up 10px, subsequent lines push it up 9px
+																								//this offsets that and keeps the first line in the same place
```
Don't do this, we don't align code, including templates.

Just put both these comments before the line itself.

cacogen commented:
I have a couple of things I need to tweak here for cyborgs

cacogen committed:
Airlock screentips play well with borg modules

cacogen committed:
Can't open UI with active module

cacogen committed:
Fixes incorrect check for aicamera and not AI

- Checks adjacency before explaining tool interactions because it just seems wrong to say "press X to do Y" and then not do that

cacogen commented:
Adjacent with active module:
https://user-images.githubusercontent.com/25089914/155448421-30a868e0-46a7-4ab6-bfa7-c4db497d848a.png
Non-adjacent with active module:
https://user-images.githubusercontent.com/25089914/155448425-c80610ef-727f-41cb-b688-324be66b1e8b.png
Adjacent with no active module:
https://user-images.githubusercontent.com/25089914/155448427-f35bfdfc-d5ed-4d0f-a6c9-32f046a948d3.png
Non-adjacent with no active module:
https://user-images.githubusercontent.com/25089914/155449131-1423a1b6-3b86-40c7-9d4a-a2587ec8770f.png

Mothblocks commented:
got damn

Mothblocks commented:
when we get rid of the stat panel, i'll ideally want to move a lot of this to the top right. a lot of that is clunky but it's not your fault, we just don't have any room left on the ui :(
